### PR TITLE
feat(CB2-12602): Display re-application date from test-results on IVA30 and MSVA30

### PR DESCRIPTION
**Pull request by Daniel-Searle**
## Display re-application date from test-results on IVA30 and MSVA30

Changed re-application date to be manually input instead of previously auto calculating.
[link to ticket number](https://dvsa.atlassian.net/browse/CB2-12602)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

File: src/services/CertificateGenerationService.ts [hunk 1]
```diff
@@ -605,7 +605,7 @@ class CertificateGenerationService {
           type: testResult.vehicleType,
           testerName: testResult.testerName,
           date: moment(testResult.testTypes.testTypeStartTimestamp).format("DD/MM/YYYY"),
-          retestDate: this.calculateVehicleApprovalRetestDate(testResult.testTypes.testTypeStartTimestamp),
+          reapplicationDate: testResult.reapplicationDate ?? "",
```
Comment (naathanbrown) [inline src/services/CertificateGenerationService.ts]: does this not need a date format like above? is it not coming as a UTC timestamp?

File: src/models/index.d.ts [hunk 2]
```diff
@@ -175,6 +175,7 @@ interface ITestResult {
   model?: string;
   bodyType?: IBodyTypeModel;
   testTypes: ITestType;
+  reapplicationDate?: string;
```
Comment (m-coslett) [inline src/models/index.d.ts]: this should in the test type model

File: src/services/CertificateGenerationService.ts [hunk 3]
```diff
@@ -589,7 +589,7 @@ class CertificateGenerationService {
           bodyType: testResult.bodyType?.description,
           date: moment(testResult.testTypes.testTypeStartTimestamp).format("DD/MM/YYYY"),
           testerName: testResult.testerName,
-          reapplicationDate: this.calculateVehicleApprovalRetestDate(testResult.testTypes.testTypeStartTimestamp),
+          reapplicationDate: testResult.reapplicationDate ?? '',
```
Comment (m-coslett) [inline src/services/CertificateGenerationService.ts]: this doesn't cover all the ac's
"need to handle that the field may not exist, may not contain a a value, or may be null or undefined"

File: src/services/CertificateGenerationService.ts (hunk 1 above)
Comment (m-coslett) [inline src/services/CertificateGenerationService.ts]: same as iva data object

File: tests/unit/certGen.unitTest.ts [hunk 4]
```diff
@@ -5199,8 +5199,8 @@ describe("cert-gen", () => {
                                 type: "motorcycle",
                                 testerName: "CVS Dev1",
                                 date: "04/03/2024",
-                                retestDate: "03/09/2024",
                                 station: "Abshire-Kub",
+                                reapplicationDate: "27/05/2024",
```
Comment (m-coslett) [inline tests/unit/certGen.unitTest.ts]: need additional tests to cover all scenarios for this date

**Review by cb-cs — DISMISSED** [review]

**Review by naathanbrown — DISMISSED** [review]

**Review by m-coslett — APPROVED** [review]

**Review by LGin-BJSS — APPROVED** [review]

**Pull request merged**